test/util/vault/util.go

```diff
@@ -144,21 +150,22 @@ func (v *VaultInitializer) CreateAppRole() (string, string, error) {
 		"period":   "24h",
 		"policies": v.role,
 	}
-	url := path.Join("/v1/auth/approle/role", v.role)
-	_, err = v.callVault("POST", url, "", params)
+
+	baseUrl := path.Join("/v1/auth", v.authPath, "role", v.role)
```

Comment:
should be `path.Join("v1", "auth", v.authPath, "role", v.role)` (i.e. let `path.Join` join v1 and auth)